pkg/proxy/proxy.go

```diff
@@ -1001,6 +1002,10 @@ func (ps *ProxySet) handleConnection(conn net.Conn, port uint32) {
 		}
 	}
 
+	mocksTotal, ok := ctx.Value("mocksTotal").(*map[string]int)
```

Comment:
Why are we doing it at this level, this will give you wrong info . Once you made a connection you can create n no of mocks and if you use this logic the result will be only 1 mock.

Reply:
You must do it at yaml level because that is when we know mocks are created as files and given to user and the count will also will be correct.

Reply:
done

Reply:
Where have you added them..?

Reply:
Let me give you an overview of we have decide . We have decided to use context to know on which mode we are on like record , test, mock-record or mock-test . With which we will emit the event in the yaml layer . Because yaml layer is actually where we create the file . So we need to do everything in the yaml layer . Even regarding the events  currently the events are according to test-sets which I feel is not correct , may in future we don't have test-sets. Events being more granular will always give better data . Like when ever a testcase is recorded we can emit an event RecordedTestAndMocks in which it will say a testcase is recorded and these mocks are recorded . We can also keep the test-set event which you have written like these many test-cases are recorded and these many mocks are recorded in test-set.